### ISSUE TITLE
Frozen POLY following a vote process is effectively burnt

### ISSUE THREAD

**Issue by adamdossa** [PolymathNetwork/polymath-core-deprecated#54]
If an owner, delegate or auditors allocation is frozen via the vote process, then those POLY tokens are effectively burnt I believe - I can’t see any way they can be extracted (e.g. by original contributors) after freezing - this is a very strong incentive not to freeze - not sure if intentional or not.

**Comment by pabloruiz55** [comment]
Maybe there could be a "voteToUnfreeze" function to revert the freeze. What do you think? 

There might be a bigger issue here. What's to stop shareholders from colluding to effectively freeze anyone's POLY just because they can? 

----

Under what circumstances would shareholders want to freeze an actor's POLY?

**Comment by satyamakgec** [comment]
I agree with @pabloruiz55. I think I have to create some function like this

**Comment by satyamakgec** [comment]
@Everhusk inputs?

**Comment by adamdossa** [comment]
I'm not sure that `voteToUnfreeze` would work well. What is the intention for frozen funds? My assumption is that allocations which are frozen after a vote should be refunded to contributors on a pro-rata basis. This would need a function `withdrawFrozenFunds` which calculates the pro-rata amount for each contributor and refunds accordingly. I think a vote, once finalised, should be in final (e.g. non-reversible).

**Comment by pabloruiz55** [comment]
The problem is that voteToFreeze looks so final.

It could be gamed, as I mentioned, by holders so if they collude they could get their money back if the token has less value than what they put.
Having this requires a lot of thinking in terms of governance and dispute resolution.

https://medium.freecodecamp.org/what-we-can-do-to-reassure-ico-investors-that-we-wont-vanish-with-their-money-ae9cfa3e162b?source=linkShare-f8f3a9df9848-1516217754

**Comment by sangheraio** [comment]
This is by design, we don't want investors to be able to have a financial incentive to freeze funds, but rather only be able to use it in extreme situations where the issuer has committed fraud (the whole purpose os securities regulations).

By burning the tokens, this could be used as a punishment mechanism for bad behaviour and also used as a vote of confidence in issuers trying to raise funds.

**Comment by adamdossa** [comment]
OK - so for now let's leave this as it is which is as an option of last resort (since investors have no economic incentive to freeze, they would need to be very upset to vote for such an outcome).

In the medium term I think reviewing the economic incentives for staking POLY for all entities is worthwhile.
